include/deal.II/matrix_free/shape_info.h

```diff
@@ -285,6 +285,35 @@ namespace internal
        * end points of the unit cell.
        */
       bool nodal_at_cell_boundaries;
+
+      /**
+       * Stores the shape values of the finite element evaluated on all
+       * quadrature points for all faces (no tensor-product structure
+       * exploited). To access the shape values for a given face `face_no` and
+       * its orientation `d`, shape_values_face_offset can be used to query a
+       * pointer into this array. The code would look like:
+       * <code>shape_values_face.data() +
+       * shape_values_face_offset[face_no][orientation]</code>.
+       */
+      AlignedVector<Number> shape_values_face;
```

Comment:
Could one also use `Table<3,Number>`? This would avoid introducing a different data structure and one could also get the appropriate data out of it.

Reply:
OK. I have introduced tables!